### PR TITLE
Cleans up some Machinery Attackby

### PR DESCRIPTION
**Pull request by Fox-McCloud**
This is largely to prep the way for expanding object damaging.

- cleans up attackby so it `return ..()`
- standardizes machinery attackby `obj/item` var casting to `I`
- uses tool helpers

### CODE REVIEW

File: code/game/machinery/adv_med.dm [hunk 1]
```diff
@@ -105,7 +104,9 @@
 		occupant = M
 		icon_state = "body_scanner_1"
 		add_fingerprint(user)
-		qdel(G)
+		qdel(TYPECAST_YOUR_SHIT)
```
Comment (KasparoVy) [inline code/game/machinery/adv_med.dm]: Dank

File: code/game/machinery/firealarm.dm [hunk 2]
```diff
@@ -62,78 +62,74 @@ FIRE ALARM
 /obj/machinery/firealarm/attack_ghost(mob/user)
 	ui_interact(user)
 
-/obj/machinery/firealarm/bullet_act(BLAH)
```
Comment (tigercat2000) [inline code/game/machinery/firealarm.dm]: Why did you remove this?

Comment (Fox-McCloud) [reply code/game/machinery/firealarm.dm]: Having custom overridden behaviors for bullet_act, and the likes should be avoided in the vast majority of instances; we want the item to be damaged, not for this to happen.

I suspect this was put in, by Bay, on the assumption of "lol, lasers are hot, it would trigger the fire alarm". It's an annoying little behavior that doesn't really fit realistically, and is going to get in the way of future behaviors (I don't think having this behavior *and* damaging it, simultaneously works well, either).

Comment (taukausanake) [reply code/game/machinery/firealarm.dm]: Well you could argue with the temperature gun that it could fuck with air alarms, but fire alarms are just switches, not sensors correct? Isn't it the air alarm that triggers containment and the fire alarm that disarms it before optimal atmos returns?

Be kind of neat to trigger alarms with matches or lighters without blowing stuff up

Comment (Fox-McCloud) [reply code/game/machinery/firealarm.dm]: They work off temperature; if the temperature is too high, then they will alarm.

File: code/game/machinery/Sleeper.dm [hunk 3]
```diff
@@ -280,84 +280,83 @@
...
-			if(src.occupant)
+		if(do_after(user, 20, target = G.affecting))
+			if(occupant)
 				to_chat(user, "<span class='boldnotice'>The sleeper is already occupied!</span>")
 				return
-			if(!GG || !GG.affecting) return
-			var/mob/M = GG.affecting
+			if(!G || !G.affecting)
+				return
+			var/mob/M = G:affecting
```
Comment (Fox-McCloud) [inline code/game/machinery/Sleeper.dm]: FIX.

File: code/game/machinery/Sleeper.dm [hunk 4]
```diff
@@ -280,84 +280,83 @@
...
+		if(occupant)
 			to_chat(user, "<span class='notice'>The scanner is occupied.</span>")
 			return
 		if(panel_open)
 			to_chat(user, "<span class='notice'>Close the maintenance panel first.</span>")
 			return
 		if(dir == 4)
 			orient = "LEFT"
-			dir = 8
+			setDir(8)
```
Comment (Crazylemon64) [inline code/game/machinery/Sleeper.dm]: Don't we have direction defines?
`1: NORTH`
`2: SOUTH`
`4: EAST`
`8: WEST`

File: code/game/machinery/Sleeper.dm [hunk 5]
```diff
@@ -280,84 +280,83 @@
...
 		else
 			orient = "RIGHT"
-			dir = 4
-		playsound(src.loc, G.usesound, 50, 1)
+			setDir(4)
```
Comment (Crazylemon64) [inline code/game/machinery/Sleeper.dm]: Direction define

File: code/game/machinery/adv_med.dm [hunk 6]
```diff
@@ -54,37 +54,36 @@
...
+	if(iswrench(I))
+		if(occupant)
 			to_chat(user, "<span class='notice'>The scanner is occupied.</span>")
 			return
 		if(panel_open)
 			to_chat(user, "<span class='notice'>Close the maintenance panel first.</span>")
 			return
 		if(dir == 4)
-			dir = 8
+			setDir(8)
```
Comment (Crazylemon64) [inline code/game/machinery/adv_med.dm]: Dir define

File: code/game/machinery/adv_med.dm [hunk 7]
```diff
@@ -54,37 +54,36 @@
...
 		else
-			dir = 4
-		playsound(src.loc, G.usesound, 50, 1)
+			setDir(4)
```
Comment (Crazylemon64) [inline code/game/machinery/adv_med.dm]: Dir define

File: code/game/machinery/adv_med.dm [hunk 8]
```diff
@@ -292,47 +293,49 @@
...
 		return
 
-	if(istype(G, /obj/item/wrench))
+	if(iswrench(I))
 		if(panel_open)
 			to_chat(user, "<span class='notice'>Close the maintenance panel first.</span>")
 			return
 		if(dir == 4)
-			dir = 8
+			setDir(8)
```
Comment (Crazylemon64) [inline code/game/machinery/adv_med.dm]: Dir define

File: code/game/machinery/gameboard.dm [hunk 9]
```diff
@@ -93,8 +93,11 @@
...
-	else if(istype(G, /obj/item/crowbar))
-		default_deconstruction_crowbar(G, ignore_panel = 1)
+/obj/machinery/gameboard/attackby(obj/item/I, mob/user, params)
+	if(default_unfasten_wrench(user, I))
+		return
+
+	if(default_deconstruction_crowbar(I, ignore_panel = TRUE))
+		return
+	else
+		return ..()
```
Comment (Crazylemon64) [inline code/game/machinery/gameboard.dm]: This is a dangerous pattern/gives off a "bad code smell", even though it can be gotten away with "safely" here - used in other situations, this could result in the parent call being missed, or being called erroneously.
What's bugging me is the `else` being tied to only one of the bits of the `if/else` chain, instead of the crowbar being an `else if` and all of them are instead bundled together
If they *were* bundled together, forgetting a `return` in the wrench block wouldn't result in parent call, and a subsequent "double-tap"

File: code/game/machinery/hologram.dm [hunk 10]
```diff
@@ -55,17 +55,20 @@ var/list/holopads = list()
...
 		return
 
-	if(default_unfasten_wrench(user, P))
+	if(default_unfasten_wrench(user, I))
 		return
 
-	default_deconstruction_crowbar(P)
+	if(default_deconstruction_crowbar(I))
+		return
+	else
```
Comment (Crazylemon64) [inline code/game/machinery/hologram.dm]: Bad pattern

File: code/game/machinery/recycler.dm [hunk 11]
```diff
@@ -60,9 +61,10 @@ var/const/SAFETY_COOLDOWN = 100
 	if(default_unfasten_wrench(user, I))
 		return
 
-	default_deconstruction_crowbar(I)
-	..()
-	add_fingerprint(user)
+	if(default_deconstruction_crowbar(I))
+		return
+	else
```
Comment (Crazylemon64) [inline code/game/machinery/recycler.dm]: bad pattern

File: code/game/machinery/requests_console.dm [hunk 12]
```diff
@@ -308,16 +309,18 @@ var/list/obj/machinery/requests_console/allConsoles = list()
...
+	if(istype(I, /obj/item/stamp))
+		if(inoperable(MAINT))
+			return
 		if(screen == RCS_MESSAUTH)
-			var/obj/item/stamp/T = O
+			var/obj/item/stamp/T = I
 			msgStamped = text("<font color='blue'><b>Stamped with the [T.name]</b></font>")
 			updateUsrDialog()
-	return
+	else
```
Comment (Crazylemon64) [inline code/game/machinery/requests_console.dm]: bad pattern

File: code/game/machinery/rechargestation.dm [hunk 13]
```diff
@@ -136,18 +136,21 @@
...
 		return
 
-	if(exchange_parts(user, P))
+	if(exchange_parts(user, I))
 		return
 
-	default_deconstruction_crowbar(P)
+	if(default_deconstruction_crowbar(I))
+		return
+	else
```
Comment (Crazylemon64) [inline code/game/machinery/rechargestation.dm]: bad pattern

File: code/game/machinery/recharger.dm [hunk 14]
```diff
@@ -54,7 +54,7 @@
 			to_chat(user, "<span class='notice'>[src] isn't connected to anything!</span>")
 		return 1
 	else
-		..()
+		return ..()
```
Comment (Crazylemon64) [inline code/game/machinery/recharger.dm]: Bad pattern

File: code/game/machinery/firealarm.dm [hunk 15]
```diff
@@ -62,78 +62,74 @@ FIRE ALARM
...
+				else if(iswrench(I))
 					to_chat(user, "<span class='warning'>You remove the fire alarm assembly from the wall!</span>")
 					new /obj/item/mounted/frame/firealarm(get_turf(user))
-					playsound(get_turf(src), W.usesound, 50, 1)
+					playsound(get_turf(src), I.usesound, 50, 1)
 					qdel(src)
-		return
 
-	alarm()
+	else
```
Comment (Crazylemon64) [inline code/game/machinery/firealarm.dm]: Bad pattern

File: code/game/machinery/dye_generator.dm [hunk 16]
```diff
@@ -50,18 +50,18 @@
...
 
-	if(istype(W, /obj/item/hair_dye_bottle))
-		user.visible_message("<span class='notice'>[user] fills the [W] up with some dye.</span>","<span class='notice'>You fill the [W] up with some hair dye.</span>")
-		var/obj/item/hair_dye_bottle/HD = W
+	if(istype(I, /obj/item/hair_dye_bottle))
+		var/obj/item/hair_dye_bottle/HD = I
+		user.visible_message("<span class='notice'>[user] fills the [HD] up with some dye.</span>","<span class='notice'>You fill the [HD] up with some hair dye.</span>")
 		HD.dye_color = dye_color
 		HD.update_dye_overlay()
 	else
```
Comment (Crazylemon64) [inline code/game/machinery/dye_generator.dm]: bad pattern

File: code/game/machinery/cloning.dm [hunk 17]
```diff
@@ -349,19 +349,19 @@
...
+	if(exchange_parts(user, I))
 		return
 
-	if(default_deconstruction_crowbar(W))
+	if(default_deconstruction_crowbar(I))
 		return
 
-	if(W.GetID())
-		if(!check_access(W))
+	if(I.GetID())
```
Comment (Crazylemon64) [inline code/game/machinery/cloning.dm]: bad pattern

File: code/game/machinery/alarm.dm [hunk 18]
```diff
@@ -1026,34 +1024,35 @@
...
+			else if(iswrench(I))
 				to_chat(user, "You remove the fire alarm assembly from the wall!")
 				new /obj/item/mounted/frame/alarm_frame(get_turf(user))
-				playsound(get_turf(src), W.usesound, 50, 1)
+				playsound(get_turf(src), I.usesound, 50, 1)
 				qdel(src)
+			return
 
-	return 0
+	return ..()
```
Comment (Crazylemon64) [inline code/game/machinery/alarm.dm]: Can this part be reached?

File: code/game/machinery/adv_med.dm [hunk 19]
```diff
@@ -292,47 +293,49 @@
...
-		playsound(loc, G.usesound, 50, 1)
+			setDir(4)
+		playsound(loc, I.usesound, 50, 1)
 
-	if(exchange_parts(user, G))
+	if(exchange_parts(user, I))
 		return
 
-	default_deconstruction_crowbar(G)
+	if(default_deconstruction_crowbar(I))
```
Comment (Crazylemon64) [inline code/game/machinery/adv_med.dm]: bad pattern

File: code/game/machinery/adv_med.dm [hunk 20]
```diff
@@ -54,37 +54,36 @@
...
 		return
 
-	if(istype(G, /obj/item/crowbar))
-		default_deconstruction_crowbar(G)
+	if(default_deconstruction_crowbar(I))
 		return
 
-	if(istype(G, /obj/item/grab))
-		var/obj/item/grab/TYPECAST_YOUR_SHIT = G
+	if(istype(I, /obj/item/grab))
```
Comment (Crazylemon64) [inline code/game/machinery/adv_med.dm]: bad pattern

File: code/game/machinery/Sleeper.dm [hunk 21]
```diff
@@ -280,84 +280,83 @@
...
 		return
 
-	if(istype(G, /obj/item/crowbar))
-		default_deconstruction_crowbar(G)
+	if(default_deconstruction_crowbar(I))
 		return
 
-	if(istype(G, /obj/item/grab))
-		var/obj/item/grab/GG = G
+	if(istype(I, /obj/item/grab))
```
Comment (Crazylemon64) [inline code/game/machinery/Sleeper.dm]: bad pattern

File: code/game/machinery/teleporter.dm [hunk 22]
```diff
@@ -341,14 +340,18 @@
...
+
+	if(exchange_parts(user, I))
 		return
 
-	if(exchange_parts(user, W))
+	if(default_deconstruction_crowbar(I))
 		return
 
-	default_deconstruction_crowbar(W)
+	else
```
Comment (Crazylemon64) [inline code/game/machinery/teleporter.dm]: bad pattern

File: code/game/machinery/teleporter.dm [hunk 23]
```diff
@@ -438,14 +441,18 @@
...
```
Comment (Crazylemon64) [inline code/game/machinery/teleporter.dm]: bad pattern

File: code/game/machinery/teleporter.dm [hunk 24]
```diff
@@ -506,40 +513,40 @@
...
-			return
-		if(istype(W, /obj/item/circuitboard/teleporter_perma))
-			var/obj/item/circuitboard/teleporter_perma/C = W
+		if(istype(I, /obj/item/circuitboard/teleporter_perma))
+			var/obj/item/circuitboard/teleporter_perma/C = I
 			C.target = teleporter_console.target
-			to_chat(user, "<span class='caution'>You copy the targeting information from \the [src] to \the [W]</span>")
-			return
+			to_chat(user, "<span class='caution'>You copy the targeting information from [src] to [C]</span>")
+	else
```
Comment (Crazylemon64) [inline code/game/machinery/teleporter.dm]: bad pattern(?)
(The `panel_open` term obscures whether this bit is deliberate)

File: code/game/machinery/Sleeper.dm (hunk 4 above)
Comment (Fox-McCloud) [reply code/game/machinery/Sleeper.dm]: Yeah, we do, I just can never remember what they are, because I don't believe they're listed anywhere (they're native BYOND defines).
